### PR TITLE
FALCON-734. Fixed the check instance part on each run to get the status

### PR DESCRIPTION
**Pull request by sandeepSamudrala**

### CODE REVIEW

File: prism/src/main/java/org/apache/falcon/service/EntitySLAMonitoringService.java [hunk 1]
```diff
@@ -362,7 +367,7 @@ void addPendingEntityInstances(Date endTime) throws FalconException {
             if (EntityType.FEED.name().equals(entityType)
                     || isEntityRunning(EntityUtil.getEntity(entityType, entityName))) {
                 Date lastMonitoredInstanceTime = monitoredEntityBean.getLastMonitoredTime();
-                Date newCheckPointTime = endTime != null ? endTime : now();
+                Date newCheckPointTime = checkPointTime;
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/service/EntitySLAMonitoringService.java]: So, checkPointTime is not expected to null. Better to add check and throw an exception.

File: prism/src/main/java/org/apache/falcon/service/EntitySLAMonitoringService.java [hunk 2]
```diff
@@ -417,12 +424,15 @@ private boolean checkEntityInstanceAvailability(String entityName, String cluste
...
                 AbstractWorkflowEngine wfEngine = WorkflowEngineFactory.getWorkflowEngine();
-                InstancesResult instancesResult = wfEngine.getStatus(entity, nominalTime, nominalTime, null, null);
-                if (instancesResult.getInstances().length > 0
-                        && instancesResult.getInstances()[0].status.equals(InstancesResult.WorkflowStatus.SUCCEEDED)){
-                    LOG.trace("Entity instance(Process:{}, cluster:{}, instanceTime:{}) is available.",
-                            entity.getName(), clusterName, nominalTime);
-                    return true;
+
+                InstancesResult instancesResult = wfEngine.getStatus(entity, nominalTime,
+                        new Date(nominalTime.getTime() + 200), PROCESS_LIFE_CYCLE, null);
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/service/EntitySLAMonitoringService.java]: Better to set the last param to false, rather than null.

Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/service/EntitySLAMonitoringService.java]: Ack. Changing the same.

**Comment by sandeepSamudrala** [comment]
I did submit and schedule a process. Stopped the falcon(active mq embedded mode) and then let the instances complete. No messages received from active mq. I started the falcon , and the succeeded instance got deleted from DB and the slaAlert api also returns the instances in accordance.

**Pull request merged**